packages/devtools_server/lib/src/devtools_command.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/command_runner.dart';
+
+import '../devtools_server.dart';
+import 'server.dart';
+
+class DevToolsCommand extends Command<int> {
+  DevToolsCommand({
+    this.customDevToolsPath,
```

Comment:
what is the customDevToolsPath for?

Reply:
In the default shelf handler, we resolve the location of package:devtools in order to find its build/ directory. For the pub published version of the package that contains the compiled web resources. In the sdk distro those don't exist relative to package:devtools but in the sdk's `bin/resources/devtools` directory; this override lets us point the devtools server to that location.